Mathlib/Algebra/Group/Subgroup/ZPowers/Lemmas.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2020 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+import Mathlib.Algebra.Group.Subgroup.ZPowers.Basic
+import Mathlib.Data.Countable.Basic
+import Mathlib.Data.Int.Cast.Lemmas
+import Mathlib.GroupTheory.Subgroup.Centralizer
+
+/-!
+# Subgroups generated by an element
+
+## Tags
+subgroup, subgroups
+
+-/
+
+variable {G : Type*} [Group G]
+variable {A : Type*} [AddGroup A]
+variable {N : Type*} [Group N]
+
+namespace Subgroup
+
+theorem range_zpowersHom (g : G) : (zpowersHom G g).range = zpowers g := rfl
```

Comment:
I'm a little surprised that this can't stay in `Basic`; can `zpowersHom` itself be moved?

Reply:
It's an `Equiv` and one of `*_inv`s depends on `AddMonoidHom.ext_int`. While I can move it, this will require more rewrites.